### DISCUSSION TITLE
How to run on windows

### DISCUSSION THREAD

**Discussion by lanttu1243** [SaptakBhoumik/Peregrine#29] [Q&A]
Whenever I try to run 
_peregrine compile test.pe -o temp.c_  
after successfully running
_cd Peregrine\Peregrine_
_v Peregrine.v -o peregrine.exe_
I get an error 
_inputString
^Undefined function_

I am wondering am I doing something wrong or is the fault in the source code.

**Comment by SaptakBhoumik** [comment]
Please clone the repo again. We have fixed this issue

**Reply by SaptakBhoumik** [reply]
@lanttu1243 Let me know if it is not working after you clone the latest repo

**Reply by lanttu1243** [reply]
@SaptakBhoumik trying to run a file manually still does not work and when running the test_runner.py  I get

Compiling ./tests\ccode.pe and running it

'.' is not recognized as an internal or external command, operable program or batch file.
gcc: error: .\temp.c: No file or directory
gcc: fatal error:  "ei sy÷tetiedostoja"
compilation interrupted
'.' is not recognized as an internal or external command, operable program or batch file.

and the same repeated for each of the files in the tests folder
the part of the error in quotation marks is in Finnish for some reason but means probably that some files do not exist.

**Reply by SaptakBhoumik** [reply]
@lanttu1243 Give me sometime. I am fixing it. It is because windows uses  `\` but unix like system uses `/`

**Reply by SaptakBhoumik** [reply]
@lanttu1243 Please clone again and run it

**Reply by lanttu1243** [reply]
I guess it now uses the right directories yet still gives most of the same errors


This program will compile the peregrine compiler and run some tests
Compiling the peregrine compiler


Compiling .\tests\ccode.pe and running it


inputString
^Undefined function
gcc: error: temp.c: No such file or directory
gcc: fatal error: ei sy÷tetiedostoja
compilation interrupted.
'.\output.exe' is not recognized as an internal or external command,
operable program or batch file.

**Reply by SaptakBhoumik** [reply]
@lanttu1243 Please check again

**Reply by lanttu1243** [reply]
still giving the same error as above.

**Reply by SaptakBhoumik** [reply]
> still giving the same error as above. 

Give me a day. Also contact me via email.   saptakbhoumik@gmail.com

**Reply by lanttu1243** [reply]
When I look into the Peregrine.v and and inserted a few println commands
// Original author: Saptak Bhoumik
fn main() {
	arg := os.args.clone()
	println(arg)
	mut filename := ''
	mut emitc := false
	mut emitast := false
	mut help := false
	mut o := false
	mut previous := ''
	mut outfile := ''
	mut idx_ := 0
	mut ext:=''
	println('Hello')
	if '${os.user_os}'=="windows"{
		ext = ".exe"
		println(ext)
	}
	for idx, x in arg {
		println(idx)
		println(x)
		idx_ = idx
		if idx == 0 {
			previous = x
			continue
		}...}
the printout is:
Compiling .\tests\function.pe and running it


['peregrine', 'compile', '.\tests\function.pe', '-emit-c']
Hello
0
peregrine
1
compile
2
.\tests\function.pe
3
-emit-c


So based on this I would guess that the if statement on line 121 is not working and it does not detect that the os is windows.

**Reply by SaptakBhoumik** [reply]
@lanttu1243 Can you print os.user_os and send me the result
